client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-survey-banner/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { useTranslate } from 'i18n-calypso';
+import { useCallback } from 'react';
+import Banner from 'calypso/components/banner';
+import { useDispatch, useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import {
+	JETPACK_DASHBOARD_SURVEY_BANNER_PREFERENCE,
+	getJetpackDashboardPreference as getPreference,
+} from 'calypso/state/jetpack-agency-dashboard/selectors';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { PreferenceType } from '../types';
+
+import './style.scss';
+
+interface Props {
+	isDashboardView?: boolean;
+}
+
+export default function SiteSurveyBanner( { isDashboardView }: Props ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const preferenceName = JETPACK_DASHBOARD_SURVEY_BANNER_PREFERENCE;
+
+	const preference = useSelector( ( state ) => getPreference( state, preferenceName ) );
+
+	const isDismissed = preference?.dismiss;
+
+	const savePreferenceType = useCallback(
+		( type: PreferenceType ) => {
+			dispatch( savePreference( preferenceName, { ...preference, [ type ]: true } ) );
+		},
+		[ dispatch, preference, preferenceName ]
+	);
+
+	const dismissAndRecordEvent = ( eventName: string ) => {
+		savePreferenceType( 'dismiss' );
+
+		const eventNamePrefix = isDashboardView
+			? 'calypso_jetpack_agency_dashboard_'
+			: 'calypso_partner_portal_';
+
+		dispatch( recordTracksEvent( eventNamePrefix + eventName ) );
```

Comment:
Could we use the `useJetpackAgencyDashboardRecordTrackEvent` hook here?

Reply:
Hi @yashwin, since we are tracking events from Partner Portal as well. I decided to use the recordTracksEvent since I can't override `useJetpackAgencyDashboardRecordTrackEvent` for our use case.